server/src/server.js

```diff
@@ -284,7 +284,7 @@ function decodeAuthHeader(header) {
   /** Decode a string header in the format {deviceId}:{deviceIdSig};abtests={b64thing}:{sig} */
   // Since it's treated as opaque, we'll use a fragile regex
   let keygrip = dbschema.getKeygrip();
-  let match = /^([^:]+):([^;]+);abTests=([^:]+):(.*)$/.exec(header);
+  let match = /^([^:]{1,225}):([^;]{1,225});abTests=([^:]{1,225}):(.{0,225})$/.exec(header);
```

Comment:
I suspect abTests could go over 255 characters (it's a base64'd JSON object), so the second-to-last one should be longer than 225. Also maybe you meant 255 instead of 225 ;) Though really they are both entirely arbitrary numbers.

Anyway, like:

`let match = /^([^:]{1,225}):([^;]{1,225});abTests=([^:]{1,1500}):(.{0,225})$/.exec(header);`